### DISCUSSION TITLE
Aplicação para gestão de Clientes e Freelancers

### DISCUSSION THREAD

**Discussion by fabioeiras8** [directus/directus#15846] [Archive]
Estou a procura de uma solução para criar uma aplicação no directus 

Que permita ao freelancer/profissional registar-se, introduzir a sua localização por escrito e essa morada seja definida num mapa.

Por sua vez a equipa do staff tb pode criar clientes e procurar freelancers mais proximos no mapa de forma a poder entegar o serviço ao freelancer.

Por sua vez o freelancer tb pode criar clientes que tenha efectuado contacto ou actualizar informação sobre o cliente.

A funcionalidade mais importante é poder fazer uma pesquisa rapida de profissionais mais proximos do cliente numa base de dados que possa ter 1000 a 5000 linhas de profissionais por exemplo

**Comment by joselcvarela** [comment]
Hello @fabioeiras8 
Let me just start with putting that into English:
Basically OP wants to be able to search the nearest records of a location using map interface.

There's already a proposal to use a `_within` operator for filter in maps:
https://github.com/directus/directus/discussions/11625

Although this feature is not implemented. If you would like to see that on Directus you can sponsor that feature in order for that being prioritized: 
https://docs.directus.io/getting-started/support.html#commissioned-features

Otherwise, you can also have what you want by getting the map limits as coordinates and then using the `_intersects` filter. 
Here you can see a user from our community using it:
https://discord.com/channels/725371605378924594/725817819882586132/1024426979635773630

And this is how you can use it in Insights module, for example, using Global Variable:

https://user-images.githubusercontent.com/14039341/193504973-ba94820d-e5c6-46b9-a37e-fa1f765c252b.mov



This video, also shows the Graphql payload you should use to get `_intersects` filter with GeoJson field

**Comment by fabioeiras8** [comment]
I have difficult to understand. Thanks your help

Cumprimentos / Best Regards

________________________________
De: José Varela ***@***.***>
Enviado: 3 de outubro de 2022 06:05
Para: directus/directus ***@***.***>
Cc: fabioeiras8 ***@***.***>; Mention ***@***.***>
Assunto: Re: [directus/directus] Aplicação para gestão de Clientes e Freelancers (Discussion #15846)


Hello @fabioeiras8<https://github.com/fabioeiras8>
Let me just start with putting that into English:
Basically OP wants to be able to search the nearest records of a location using map interface.

There's already a proposal to use a _within operator for filter in maps:
#11625<https://github.com/directus/directus/discussions/11625>

Although this feature is not implemented. If you would like to see that on Directus you can sponsor that feature in order for that being prioritized:
https://docs.directus.io/getting-started/support.html#commissioned-features

Otherwise, you can also have what you want by getting the map limits as coordinates and then using the _intersects filter.
Here you can see a user from our community using it:
https://discord.com/channels/725371605378924594/725817819882586132/1024426979635773630

And this is how you can use it in Insights module, for example, using Global Variable:

https://user-images.githubusercontent.com/14039341/193504973-ba94820d-e5c6-46b9-a37e-fa1f765c252b.mov

This video, also shows the Graphql payload you should use to get _intersects filter with GeoJson field
